marklogic-data-hub-central/src/main/resources/explore-data/ml-search-lib/search.sjs

```diff
@@ -34,7 +34,18 @@ class Search {
     let facets = searchParams.selectedFacets;
     let keys = Object.keys(facets);
     let constraintArr = [];
-    keys.forEach(key => constraintArr.push(facets[key].map(value => key+':' + '"' + value + '"').join(" OR ")))
+    keys.forEach(key => {
+
+      if(xdmp.castableAs("http://www.w3.org/2001/XMLSchema", "date", facets[key][0])) {
```

Comment:
@AkshayBajajML and @rahulvudutala, we need to consider this API update in the context of the other facet selection types we will need to support, e.g.:
<img width="416" alt="Screen Shot 2022-04-01 at 5 25 00 PM" src="https://user-images.githubusercontent.com/477757/161356693-ad73bfe9-4be0-4885-8741-7336cadbdf3b.png">
Will we also be able to pass in:
- A SINGLE date or datatime to support a single (non-range) date facet?
- A range of two NUMERIC values to support a numeric range facet?

@rahulvudutala, would it be straightforward to support these two other cases with this PR?

Reply:
@wooldridge It works with date and dateTime formats. I don't think it will work with numeric formats. We are not passing the data types in the search request. Validation is done to check if the string is a date or dateTime.
If the year is only expected to be passed, then it needs to convert into either date or dateTime formats before the request is sent. 

For example, 
1) if 1974 - 1995 is selected, then it needs to be converted to 1974-01-01 and 1995-12-31
2) If only 1995 is selected, then it needs to be converted to 1995-01-01 and 1995-12-31